lib/sqlitex.ex

```diff
@@ -51,12 +48,47 @@ defmodule Sqlitex do
     exec(db, stmt)
   end
 
+  defp bind_and_query(statement, opts) do
+    {params, into} = query_options(opts)
+    case :esqlite3.bind(statement, params) do
+      {:error, _}=error -> error
+      :ok ->
+        types = :esqlite3.column_types(statement)
+        columns = :esqlite3.column_names(statement)
+        rows = :esqlite3.fetchall(statement)
+        return_rows_or_error(types, columns, rows, into)
+    end
+  end
+
   defp query_options(opts) do
-    params = Keyword.get(opts, :bind, [])
+    params = opts |> Keyword.get(:bind, []) |> translate_bindings
     into = Keyword.get(opts, :into, [])
     {params, into}
   end
 
+  defp translate_bindings(params) do
+    Enum.map(params, fn
+      nil -> :undefined
+      true -> 1
+      false -> 0
+      {{_yr, _mo, _da}, {_hr, _mi, _se}}=datetime -> datetime_to_string(datetime)
+      {{yr, mo, da}, {hr, mi, se, _usecs}} -> datetime_to_string({{yr, mo, da}, {hr, mi, se}})
```

Comment:
if Ecto is standardizing around `{hr, mi, se, usecs}` maybe we should return that form when we query and use that form when we are binding?  I picked `{{yr,mo,da},{hr,mi,se}}` because that is the format I saw in most of the [erlang documentation](http://www.erlang.org/doc/man/erlang.html#localtime-0), but since we are wrapping around `esqlite3` anyway, maybe it makes more sense for us to stick to elixir convention rather than erlang conventions?